qiskit_experiments/calibration_management/save_utils.py

```diff
@@ -0,0 +1,253 @@
+    """
+    # This can dispatch canonicalize function as version evolves.
+    model = _canonicalize_calibration_data_v1(cals, most_recent_only)
+    return asdict(model)
+
+
+def from_dict(
+    cal_data: Dict[str, Any],
+) -> Calibrations:
+    """A helper function to build calibration instance from canonical dictionary.
+
+    Args:
+        cal_data: Calibration data that conforms to the predefined data schema.
+
+    Returns:
+        Calibration instance.
+
+    Raises:
+        ValueError: When input data model version is no longer supported.
+        KeyError: When input data model doesn't conform to data schema.
+    """
+    # Apply schema for data field validation
+    try:
+        version = cal_data["schema_version"]
+        if version == "1.0":
+            model = CalibrationModelV1.apply_schema(cal_data)
+        else:
+            raise ValueError(
+                f"Loading calibration data with schema version {version} is no longer supported. "
+                "Use the same version of Qiskit Experiments at the time of saving."
+            )
+    except (KeyError, TypeError) as ex:
+        raise KeyError(
+            "Loaded data doesn't match with the defined data schema. "
+            "Check if this object is dumped from the Calibrations instance."
+        ) from ex
+
+    # This can dispatch loading mechanism depending on schema version
+    cals = Calibrations(
+        coupling_map=model.device_coupling_graph,
+        control_channel_map=model.control_channel_map.chan_map,
+        backend_name=model.backend_name,
+        backend_version=model.backend_version,
+    )
+
+    # Add schedules
+    for sched in model.schedules:
+        cals.add_schedule(
+            schedule=sched.data,
+            qubits=sched.qubits if len(sched.qubits) != 0 else None,
+            num_qubits=sched.num_qubits,
+        )
+
+    # Add parameters
+    for param in model.parameters:
+        param_value = ParameterValue(
+            value=param.value,
+            date_time=param.datetime,
+            valid=param.valid,
+            exp_id=param.exp_id,
+            group=param.group,
+        )
+        cals.add_parameter_value(
+            value=param_value,
+            param=param.param_name,
+            qubits=tuple(param.qubits),
+            schedule=param.schedule,
+            update_inst_map=False,
+        )
+    cals.update_inst_map()
+
+    return cals
+
+
+def _canonicalize_calibration_data_v1(
+    cals: Calibrations,
+    most_recent_only: bool = True,
+) -> CalibrationModelV1:
+    """A helper function to canonicalize calibration data.
+
+    Args:
+        cals: Calibration instance to apply data model.
+        most_recent_only: Set True to save calibration parameters with most recent time stamps.
+
+    Returns:
+        Canonical calibration data.
+    """
+    schedules = getattr(cals, "_schedules")
+    num_qubits = getattr(cals, "_schedules_qubits")
+    parameters = getattr(cals, "_params")
```

Comment:
Accessing the private variables like this suggests that this function should be in Calibrations (which is already somewhat long). Are there other solutions? Otherwise I'd be willing to keep this logic here.

Reply:
I don't have good alternative. You can feel free to modify this logic when you update implementation of Calibrations. I realized that we don't need to keep write function of a particular version, because it doesn't make sense to write calibration instance with different model version, i.e. Calibration model of different version is for Calibrations instance at different QE version. Supporting data load for different version is still important, but when you write an instance you don't have Calibrations instance of different QE version in your computer memory.

Reply:
Ok, let's leave as is for now.